src/app/application.cpp

```diff
@@ -403,14 +403,12 @@ void Application::runExternalProgram(const BitTorrent::Torrent *torrent) const
     LogMsg(tr("Torrent: %1, running external program, command: %2").arg(torrent->name(), program));
 
 #if defined(Q_OS_WIN)
-    auto programWchar = std::make_unique<wchar_t[]>(program.length() + 1);
-    program.toWCharArray(programWchar.get());
-
     // Need to split arguments manually because QProcess::startDetached(QString)
     // will strip off empty parameters.
     // E.g. `python.exe "1" "" "3"` will become `python.exe "1" "3"`
     int argCount = 0;
-    std::unique_ptr<LPWSTR[], decltype(&::LocalFree)> args {::CommandLineToArgvW(programWchar.get(), &argCount), ::LocalFree};
+    std::unique_ptr<LPWSTR[], decltype(&::LocalFree)> args {::CommandLineToArgvW(
+        program.toStdWString().c_str(), &argCount), ::LocalFree};
```

Comment:
It is incorrect to do `toStdWString().c_str()`. You produce dangling pointer.

Reply:
Ok, fixed.

Reply:
> It is incorrect to do `toStdWString().c_str()`. You produce dangling pointer.

It looks like I've forgotten something from theory and confused you:
>All temporary objects are destroyed as the last step in evaluating the full-expression that (lexically) contains the point where they were created, and if multiple temporary objects were created, they are destroyed in the order opposite to the order of creation

So it is still correct to have such an expression as a function argument.